packages/world/test/KeysInTableModule.t.sol

```diff
@@ -0,0 +1,200 @@
+
+    // Get the list of keys in the target table
+    keysInTable = KeysInTable.get(world, targetTableId);
+
+    // Assert that the list is correct
+    assertEq(keysInTable.length, 2, "7");
+    assertEq(keysInTable[0], key1, "8");
+
+    // Delete the first key
+    // !gasreport delete a record on a table with keysInTableModule installed
+    world.deleteRecord(namespace, sourceFile, keyTuple1);
+
+    // Get the list of keys in the target table
+    keysInTable = KeysInTable.get(world, targetTableId);
+
+    // Assert that the list is correct
+    assertEq(keysInTable.length, 1, "9");
+    assertEq(keysInTable[0], key2, "10");
+  }
+
+  function testSetField(uint256 value1, uint256 value2) public {
+    _installkeysInTableModule();
+
+    // Set a value in the source table
+    // !gasreport set a field on a table with keysInTableModule installed
+    world.setField(namespace, sourceFile, keyTuple1, 0, abi.encodePacked(value1));
+
+    // Get the list of keys in the target table
+    bytes32[] memory keysInTable = KeysInTable.get(world, targetTableId);
+
+    // Assert that the list is correct
+    assertEq(keysInTable.length, 1);
+    assertEq(keysInTable[0], key1);
+
+    // Change the value using setField
+    // !gasreport change a field on a table with keysInTableModule installed
+    world.setField(namespace, sourceFile, keyTuple1, 0, abi.encodePacked(value2));
+
+    // Get the list of keys in the target table
+    keysInTable = KeysInTable.get(world, targetTableId);
+
+    // Assert that the list is correct
+    assertEq(keysInTable.length, 1);
+
+    // Get the list of keys in the target table
+    keysInTable = KeysInTable.get(world, targetTableId);
+
+    // Assert that the list is correct
+    assertEq(keysInTable.length, 1);
+    assertEq(keysInTable[0], key1);
+  }
+
+  function testGetTargetTableSelector() public {
+    // !gasreport compute the target table selector
+    bytes32 targetTableSelector = getTargetTableSelector(MODULE_NAMESPACE, sourceTableId);
+
+    // The first 8 bytes are the module namespace
+    assertEq(bytes8(targetTableSelector), MODULE_NAMESPACE);
+
+    // followed by the first 4 bytes of the source table namespace
+    assertEq(bytes8(targetTableSelector << 64), bytes8(namespace));
+
+    // The last 16 bytes are the source file
+    assertEq(targetTableSelector.getName(), sourceFile);
+  }
+
+  function testGetkeysInTable(uint256 value1) public {
+    _installkeysInTableModule();
+
+    // Set a value in the source table
+    world.setRecord(namespace, sourceFile, keyTuple1, abi.encodePacked(value1));
+
+    // !gasreport Get list of keys in a given table
+    bytes32[] memory keysInTable = getKeysInTable(world, sourceTableId);
+
+    // Assert that the list is correct
+    assertEq(keysInTable.length, 1);
+    assertEq(keysInTable[0], key1);
+
+    // Set another key with the same value
+    world.setRecord(namespace, sourceFile, keyTuple2, abi.encodePacked(value1));
```

Comment:
Nit - I feel like setting a different value here would represent the purpose of this module slightly better (if we set the same value the difference to `KeysWithValue` is a bit less clear)